File: 백준/Silver/1406. 에디터/에디터.py
import sys

left = list(input().strip())
tc = int(input())

right = []

for _ in range(tc):
  op = sys.stdin.readline().split()
  if op[0] == 'L' and left:  # 왼쪽 이동
    right.append(left.pop())

  elif op[0] == 'D' and right:  # 오른쪽 이
    left.append(right.pop())

  elif op[0] == 'B' and left:  # 커서 왼쪽 문자 삭제
    left.pop()

  elif op[0] == 'P':
    left.append(op[1])

print(''.join(left), ''.join(right[::-1]), sep="")

File: 백준/Silver/1406. 에디터/에디터.cc
// 1406
#include <bits/stdc++.h>
using namespace std;

int main() {
  string s = ""; // 시작 문자
  int m;
  stack<char> left;  // 왼쪽 스택
  stack<char> right; // 오른쪽 스택

  cin >> s;
  cin >> m;

  // 문자열 스택 초기화
  for (int a = 0; a < (int)s.size(); a++) {
    left.push(s[a]);
  }
  // 탐색 시작
  char order, c;
  for (int i = 0; i < m; i++) {
    cin >> order;
    if (order == 'L') {
      if (!left.empty()) {
        right.push(left.top());
        left.pop();
      }
    } else if (order == 'D') {
      if (!right.empty()) {
        left.push(right.top());
        right.pop();
      }
    } else if (order == 'B') {
      if (!left.empty())
        left.pop();
    } else {
      cin >> c;
      left.push(c);
    }
  }

  // 스택 안의 문자들 합치기
  // 1. 왼쪽 스택 문자들 전부 오른쪽으로 이동
  while (!left.empty()) {
    right.push(left.top());
    left.pop();
  }
  // 2. 오른쪽 스택에서 출력
  while (!right.empty()) {
    cout << right.top();
    right.pop();
  }
}